include/cograph_rec/Cotree.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <graph/GraphTools.hpp>
+
+namespace Koala {
+    class Cotree {
+    public:
+        NetworKit::Graph *graph;
+        std::vector<long long> left_son, right_son, parent, type, size, color, number_of_colors;

```

Comment:
It's probably better to create an auxiliary structure CoNode with fields left, right, parent, type and define std::vector<CoNode> here (if it's a vector of structs, they are also allocated as chunks of continuous memory, unlike vector of pointers, so it's fast)